### DISCUSSION TITLE
unseen data evaluation

### DISCUSSION THREAD

**Discussion by ArianKl** [pycaret/pycaret#2313] [Q&A]
Hello PyCaret team, 
Thank you very much for creating this wonderful library. It's so easy to use and I love it. 

I have a question on the evaluation of model with the unseen data. 
I had a sample of 50 patients and using pycaret classification module I was able to get a good accuracy of 85% in classifying patients based on their outcome. Then I tested it on an independent test data (unseen) and again achieved a good accuracy. I was trying to plot some figures for the unseen data, but couldn't produce it. Below is the code for the unseen predictions.
unseen_predictions = predict_model(final_model, data=data_unseen)

Can anyone please tell me how to plot the figures for the unseen data?
To be precise, I used plot_model(final_model, plot = 'confusion_matrix') to create a confusion matrix, plot_model(final_model, plot = 'auc', scale = 2) to create ROC etc.

But I couldn't figure out a way to plot it for the unseen data. Please help!
I look forward to hearing from the experts.
Thank you,
AK

**Comment by kilickursat** [comment]
Hi,
You may try to create a model with unseen data.

Here is my suggestion;

```python
session_1 = setup(data_unseen, target = 'ABC', session_id=123, log_experiment=False, experiment_name='SOMETHING', normalize_method="minmax")
best_model = compare_models()


plot_model(NAME OF MODEL, plot = 'confusion_matrix')
plot_model(NAME OF MODEL, plot = 'AUC')
```
